### PR TITLE
fix imshow when image is RGBA

### LINKED ISSUE #1898: imshow don't work with RGBA image when transform keyword is provided
### Description
Hi,

When using imshow with RGBA data, and with **no transformation** (*PlateCarre* data to *PlateCarre* axis for example), there is no problem.

However, **when using transformation** (for example, *PlateCarre* data to *Orthographic* axis), the A channel from RGBA data is skipped.

#### Code to reproduce and minimal example

```python
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

fig = plt.figure(figsize=(8,5), dpi=120)
ax1 = fig.add_subplot(2, 2, 1, projection=ccrs.Orthographic(-10, 45))
ax2 = fig.add_subplot(2, 2, 2, projection=ccrs.Orthographic(-10, 45))
ax3 = fig.add_subplot(2, 2, 3, projection=ccrs.PlateCarree())
ax4 = fig.add_subplot(2, 2, 4, projection=ccrs.PlateCarree())

dy, dx = (9,18)
extent = [-180, 180, -90, 90]

# create RGB Image with random data
img1 = np.random.randint(0, 255, size=(dy, dx, 3), dtype=np.uint8)
for ax in ax1, ax3:
    ax.imshow(img1, extent=extent, transform=ccrs.PlateCarree())
    ax.set_title("no alpha (RGB)")

# Create RGBA Image with random data and linspace alpha
img2 = np.zeros((dy, dx, 4), dtype=np.uint8)
img2[:,:,:3] = img1
img2[:,:,3] = np.linspace(0, 255, dx, dtype=np.uint8).reshape(1,dx)

for ax in ax2, ax4:
    ax.imshow(img2, extent=extent, transform=ccrs.PlateCarree())
    ax.set_title("with alpha (RGBA)")

fig.suptitle("Problem imshow with alpha")
fig
```
image

With cartopy==0.18 there is no problems : 

image

I noticed this problem using cartopy 0.19.x and 0.20.x . I suspect that #1582 introduced this problem.

after reading commit changes, i noticed i did not use alpha kwargs, so i tried it : 

```python
fig = plt.figure(figsize=(12,5), dpi=120)
ax1 = fig.add_subplot(2, 3, 1, projection=ccrs.Orthographic(120, 45))
ax2 = fig.add_subplot(2, 3, 2, projection=ccrs.Orthographic(120, 45))
ax3 = fig.add_subplot(2, 3, 3, projection=ccrs.Orthographic(120, 45))
ax4 = fig.add_subplot(2, 3, 4, projection=ccrs.PlateCarree())
ax5 = fig.add_subplot(2, 3, 5, projection=ccrs.PlateCarree())
ax6 = fig.add_subplot(2, 3, 6, projection=ccrs.PlateCarree())

dy, dx = (9,18)
extent = [-180, 180, -90, 90]

# create RGB Image with random data
img1 = np.random.randint(0, 255, size=(dy, dx, 3), dtype=np.uint8)
for ax in ax1, ax4:
    ax.imshow(img1, extent=extent, transform=ccrs.PlateCarree())
    ax.set_title("no alpha (RGB)")

# Create RGBA Image with random data and linspace alpha
img2 = np.zeros((dy, dx, 4), dtype=np.uint8)
img2[:,:,:3] = img1
img2[:,:,3] = np.linspace(0, 255, dx, dtype=np.uint8).reshape(1,dx)

for ax in ax2, ax5:
    ax.imshow(img2, extent=extent, transform=ccrs.PlateCarree())
    ax.set_title("with alpha (RGBA)")

    
for ax in ax3, ax6:
    # converting 0...255 values to 0...1 float, or exception is raised
    ax.imshow(img2, alpha=img2[:,:,3]/255, extent=extent, transform=ccrs.PlateCarree())
    ax.set_title("with alpha (RGBA) and alpha kwargs")

fig.suptitle("Problem imshow with alpha")
fig
```

This gives interesting results. Only the data where  alpha=1 is plotted, others are skipped

image

I'll continue investing next days

<details>
 <summary>Full environment definition</summary>

### Cartopy version

```python
import cartopy
import matplotlib

print("cartopy    : ", cartopy.__version__)
print("matplotlib : ", matplotlib.__version__)
```

```bash
cartopy    :  0.19.0
matplotlib :  3.4.3
```

</details>

### PR DESCRIPTION
**Pull request by ludwigVonKoopa**
Hello ! It is my first PR on cartopy. I apologize if i did something wrong or forgot  something

## Problem

tldr : fix #1898

When using imshow with RGBA data, when data projection and axis projection are the same, there is no problems.

But when there is reprojection, the A (alpha channel) is skipped and forced to 1. 

```python
import cartopy
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
import numpy as np
np.random.seed(0)

dy, dx = (4,10)

# Create RGBA Image with random data and linspace alpha
RGBA = np.zeros((dy, dx, 4), dtype=np.uint8)
RGBA[:,:,:3] = np.random.randint(0, 255, size=(dy, dx, 3), dtype=np.uint8)
RGBA[:,:,3]  = np.linspace(0, 255, dx, dtype=np.uint8).reshape(1,dx)

def plot():
    fig  = plt.figure(figsize=(8,3), dpi=120)

    ax1  = fig.add_subplot(1, 3, 1,  projection=ccrs.Orthographic(central_latitude=45))
    ax1.set_title("Orthographic axis")
    ax2  = fig.add_subplot(1, 3, 2,  projection=ccrs.PlateCarree())
    ax2.set_title("PlateCarree axis")
    ax3  = fig.add_subplot(1, 3, 3)
    ax3.set_title("no projection axis")
    return fig, ax1, ax2, ax3
```

```python
fig, ax1, ax2, ax3 = plot()
ax1.imshow(RGBA, transform=ccrs.PlateCarree())
ax2.imshow(RGBA, transform=ccrs.PlateCarree())
ax3.imshow(RGBA, )
   
fig.suptitle("RGBA u1 data in PlateCarre projection")
```
image

moreover, when `alpha` keyword is used, it completely broke data : 

```python
fig, ax1, ax2, ax3 = plot()
ax1.imshow(RGBA, alpha=0.3, transform=ccrs.PlateCarree())
ax2.imshow(RGBA, alpha=0.3, transform=ccrs.PlateCarree())
ax3.imshow(RGBA, alpha=0.3, )
   
fig.suptitle("RGBA u1 data in PlateCarre projection with alpha kwargs")
```

image

but when RGBA image is np.float64 dtype (last example was np.uint8), the alpha channel works, but outlast alpha channel from image : 

```python
fig, ax1, ax2, ax3 = plot()
ax1.imshow(RGBA/255, alpha=0.3, transform=ccrs.PlateCarree())
ax2.imshow(RGBA/255, alpha=0.3, transform=ccrs.PlateCarree())
ax3.imshow(RGBA/255, alpha=0.3)
   
fig.suptitle("RGBA f4 data in PlateCarre projection with alpha kwargs")
```
image

## Implications

when reprojection is done, it should follow matplotlib usage for alpha.

## Same plot with corrections : 

image
image
image

* test created & OK
* modifications don't broke fix introduced by #1570

### CODE REVIEW

**Comment by ludwigVonKoopa** [comment]
I see "SciTools-CLA-checker - CLA doesn't exist for all commits" but i don't understand what to do to correct this.
How can i correct this error ?
Thanks

**Comment by ludwigVonKoopa** [comment]
Ok i have signed the CLA in https://scitools.org.uk/.
